infrastructure/lib/stacks/gitHubWorkflowMonitorAlarms.ts

```diff
@@ -0,0 +1,48 @@
+/**
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+import {Duration, Stack} from "aws-cdk-lib";
+import {Construct} from "constructs";
+import {Alarm, ComparisonOperator, Metric, TreatMissingData} from "aws-cdk-lib/aws-cloudwatch";
+
+export interface AlarmProps {
+    readonly namespace: string;
+    readonly metricName: string;
+    readonly workflows: string[];
+}
+
+export class GitHubWorkflowMonitorAlarms extends Stack {
+    readonly workflowAlarmsArn: string[] = [];
+    readonly metricName: string
+    constructor(scope: Construct, id: string, props: AlarmProps) {
+        super(scope, id);
+        props.workflows.forEach(workflow => {
+            const dimensionValue = workflow;
+            const workflowMetric = new Metric({
+                namespace: props.namespace,
+                metricName: props.metricName,
```

Comment:
I assume this metric would already be flowing into CW through the general workflow, right?

Reply:
Ignore, this is just creating a metric and data will flow later once this metric object is created.

Reply:
That is correct Rishab, coming from this PR https://github.com/opensearch-project/automation-app/pull/21 the data will flow from the app events.